File: LeetCode/0461. Hamming Distance/solution.cpp
/**
 * 149 / 149 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.7 MB 
 */
class Solution {
public:
    int hammingDistance(int x, int y) {
        return __builtin_popcount(x ^ y);
    }
};

File: LeetCode/0461. Hamming Distance/solution.py
"""
56 / 56 test cases passed.
Runtime: 28 ms
Memory Usage: 14.7 MB
"""
class Solution:
    def hammingDistance(self, x: int, y: int) -> int: 
        return str(bin(x ^ y)).count('1')

"""
56 / 56 test cases passed.
Runtime: 44 ms
Memory Usage: 14.7 MB
"""
class Solution2:
    def hammingDistance(self, x: int, y: int) -> int: 
        ans, n = 0, x ^ y
        while n:
            ans += n & 1
            n >>= 1
        return ans
